### PR TITLE
Adding new field in device info, about status of HSM on the device

### PR DESCRIPTION
**Pull request by cshari-zededa**
Signed-off-by: Hariharasubramanian C S <cshari@zededa.com>

### CODE REVIEW

File: api/proto/info/info.proto [hunk 1]
```diff
@@ -208,6 +208,7 @@ message ZInfoDevice {
 
   SystemAdapterInfo systemAdapter = 24;
   uint32 restartCounter = 25; // Number of times zedagent has restarted i.e., device reboot
+  string HwSecurityModule = 26; //State of hardware security modules, like TPM
```
Comment (eriknordmark) [inline api/proto/info/info.proto]: Why not make this an enum, with values like "Unknown", "NotFound", "Disabled", and "Enabled"?
Then add a separate string with HwSecurityModuleName which is an informational string about the hardware

File: pkg/pillar/cmd/zedagent/handlemetrics.go [hunk 2]
```diff
@@ -1115,6 +1115,10 @@ func PublishDeviceInfoToZedCloud(ctx *zedagentContext) {
 
 	ReportDeviceInfo.RestartCounter = ctx.restartCounter
 
+	//Operational information about TPM presence/absence/usage.
+	//"Absent" for now, till we enable TPM functionality.
+	ReportDeviceInfo.HwSecurityModule = "Absent"
```
Comment (eriknordmark) [inline pkg/pillar/cmd/zedagent/handlemetrics.go]: Presumably "Unknown" makes more sense initially.

File: api/proto/info/info.proto (hunk 1 above)
Comment (cshari-zededa) [reply api/proto/info/info.proto]: @nordmark Updated the PR, please have a look.

File: pkg/pillar/cmd/zedagent/handlemetrics.go (hunk 2 above)
Comment (cshari-zededa) [reply pkg/pillar/cmd/zedagent/handlemetrics.go]: @nordmark Updated the PR, please have a look.

**Pull request merged**